azkaban-common/src/main/java/azkaban/imagemgmt/permission/PermissionManagerImpl.java

```diff
@@ -77,4 +78,34 @@ public boolean hasPermission(final UserManager userManager, final String imageTy
     }
     return ownerSet.contains(userId) || userManager.validateUserGroupMembership(userId, ownerSet);
   }
+
+  /**
+   * Checks the user permission based on image type name, user id and Permission type;
+   * Azkaban admin would have permission and other user validates through image_ownership table,
+   *
+   *
+   * @param imageTypeName
+   * @param user
+   * @return boolean
+   */
+  @Override
+  public Set<String> validatePermissionAndGetOwnerships(String imageTypeName, User user)
+      throws ImageMgmtException {
+    String userId = user.getUserId();
+    // validate azkaban admin
+    boolean isAzkabanAdmin = user.getRoles().stream()
+        .anyMatch(role -> userManager.getRole(role).getPermission().isPermissionSet(Type.ADMIN));
```

Comment:
Can you move LoginAbstractAzkabanServlet's isAzkabanAdmin into this class and reuse that method here to prevent the code redundancy?

Reply:
ok, this makes sense.

Reply:
I still dont see code dedup or am I missing something?

Reply:
check the latest code, not this version.
For review purpose I didn't squash all the commits. But every time the changes based on the review would be a separate commit.